src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -370,7 +375,20 @@ public void HandleIncomingRequest(Message message, ActivationData targetActivati
         {
             lock (targetActivation)
             {
-                if (targetActivation.State == ActivationState.Invalid)
+                if (targetActivation.Grain.IsGrain)
+                {
+                    var request = ((InvokeMethodRequest)message.GetDeserializedBody(this.serializationManager));
```

Comment:
There are optimizations which could be made by deferring deserialization of the args until the invoker: currently we use and `object[]`, which means that every arg has its type serialized in full. By deferring deserialization, we can take advantage of the expected argument types. It would take a little work, but if you want to make changes there anyway then it might be worth it.

Bear in mind that `InvokeMethodRequest` is a public type, though